docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,280 @@ method_summary   ::= "ALWAYS" "(" value ")"
+from CVL do not need to explicitly pass an {term}`environment` value as the
+first argument.  Second, the prover will verify that the method implementation
+in the contract being verified does not depend on any of the environment
+variables.  The results of this check are displayed on the verification report
+as a separate rule called `envfreeFuncsStaticCheck`.
+
 ```{todo}
-This document is incomplete.  See {doc}`/docs/confluence/advanced/methods-overview`,
-{doc}`/docs/confluence/advanced/summaries`, {doc}`/docs/confluence/advanced/internal-summaries`,
-and {doc}`/docs/confluence/advanced/expressive-summaries` for the old documentation
-about the methods block.
+There is a separate check called `envfreeFuncsAreNonpayable`.  Why is this
+necessary?
 ```
 
-## Non-summary declarations
+Finally, the method entry may contain an optional summarization (indicated by
+`=>` followed by the summary type and an optional application policy).  A
+summarized declaration indicates that the prover should replace some calls to
+the summarized function by an approximation.  This is an important technique
+for working around prover timeouts and also for working with external contracts
+whose implementation is not fixed at verification time.
 
-(envfree)=
-### The envfree modifier
+The summary type determines what type of approximation is used to replace the
+function calls.  The available types are described in the following sections:
+
+ * {ref}`const-summary`
+ * {ref}`havoc-summary`
+ * {ref}`dispatcher`
+ * {ref}`auto-summary`
+ * {ref}`function-summary`
+
+The application policy determines which function calls are replaced by
+approximations.  See {ref}`summaries` for details.
 
 (summaries)=
-## Summary declarations
+Summarized function calls
+-------------------------
+
+Whether a function call is replaced by an approximation depends on the context
+in which the function is called in addition to the application policy for its
+signature.  If present, the application policy must be either `ALL` or
+`UNRESOLVED`; the default policy is TODO.  The decision is made as follows:
+
+ * If the function is called from CVL rather than from contract code then it is
+   never replaced by a summary.
+
+ * If the code for the function is known at verification time, either because
+   it is a method of `currentContract` or because the receiver contract is
+   {ref}`linked <linking>`, then the function is only summarized if the
+   resolution type is `ALL`.
 
-### Application policies (UNRESOLVED or ALL)
+ * If the code for the function is not known at verification time, then the
+   function call must be summarized.  If no summary is given, the default summary
+   type is {ref}`AUTO <auto-summary>`, whose behavior is determined by the type of
+   function call.  In this case, the verification report will contain a contract
+   call resolution warning.
 
-### `ALWAYS`, `CONSTANT`, `PER_CALLEE_CONSTANT`
+```{todo}
+The default application policy is currently undocumented.
```

Comment:
The default policy is `ALL` with the exception of a dispatcher summary whose default policy is `UNRESOLVED`.

Reply:
fixed.